test/FunctionalTests/Microsoft.OData.Core.Tests/UriParser/Parsers/UriPathParserTests.cs

```diff
@@ -198,68 +198,6 @@ public void ParsePathIgnoresQueryStringEvenWhenItHasSlashes()
-        [InlineData("root:/abc:/property:/", new[] { "root", ":abc:", "property", ":" })]
-        [InlineData("root:/abc:/property:/:", new[] { "root", ":abc:", "property", "::" })]
-        [InlineData("root:/abc:/property:/::/", new[] { "root", ":abc:", "property", "::", ":" })]
-        [InlineData("root:/:/property", new[] { "root", "::", "property" })]
-        [InlineData("root:/photos/2018/February", new[] { "root", ":photos/2018/February" } )]
-        [InlineData("root:/photos%2f2018%2fFebruary", new[] { "root", ":photos%2f2018%2fFebruary" } )]
-        [InlineData("root:/photos%2f2018%2fFebruary/Others", new[] { "root", ":photos%2f2018%2fFebruary/Others" } )]
-        [InlineData("root:/photos%2f2018%2f/////February/Others", new[] { "root", ":photos%2f2018%2f/////February/Others" } )]
-        [InlineData("root:/photos/2018%2fFebruary:/permissions", new[] { "root", ":photos/2018%2fFebruary:", "permissions" } )]
-        [InlineData("root:/photos:2018:/permissions:/abc", new[] { "root", ":photos:2018:", "permissions", ":abc" } )]
-        [InlineData("root:/photos::::::2018:/permissions:/abc", new[] { "root", ":photos::::::2018:", "permissions", ":abc" })]
-        [InlineData("root:/photos/::::::2018:/permissions:/abc", new[] { "root", ":photos/::::::2018:", "permissions", ":abc" })]
-        [InlineData("EntitySet('key'):/xyz:/perm", new[] { "EntitySet('key')", ":xyz:", "perm" } )]
-        [InlineData("EntitySet('key'):/xyz::/perm", new[] { "EntitySet('key')", ":xyz:", ":perm" } )]
-        [InlineData("EntitySet('key'):/xyz/abc:", new[] { "EntitySet('key')", ":xyz/abc:" } )]
-        [InlineData("EntitySet('key%2fvalue'):/xyz", new[] { "EntitySet('key%2fvalue')", ":xyz" })]
-        [InlineData("EntitySet/key%2fvalue:/xyz", new[] { "EntitySet", "key%2fvalue", ":xyz" })]
-        [InlineData("EntitySet/key:%2fvalue:/xyz", new[] { "EntitySet", "key:%2fvalue", ":xyz" })]
-        [InlineData("EntitySet('key'):/xyz::/perm", new[] { "EntitySet('key')", ":xyz:", ":perm" })]
-        [InlineData("EntitySet('key'):/xyz:/:/perm", new[] { "EntitySet('key')", ":xyz:", ":perm" })]
-        [InlineData(":/xyz::/perm", new[] { ":xyz:", ":perm" })]
-        [InlineData(":/xyz:/:/perm", new[] { ":xyz:", ":perm" })]
-        [InlineData(":/xyz://///:/perm", new[] { ":xyz:", ":perm" })]
-        public void ParsePathShouldAllowEscapeFunctionPattern(string pattern, string[] expected)
-        {
-            // Arrange
-            var fullUrl = new Uri(this.baseUri.AbsoluteUri + pattern);
-
-            // Act
-            var actual = this.pathParser.ParsePathIntoSegments(fullUrl, this.baseUri);
-
-            // Assert
-            Assert.Equal(expected, actual);
-        }
-
-        [Theory]
-        [InlineData("root::/abc", "root::")]
-        [InlineData("EntitySet('key')::/abc", "EntitySet('key')::")]
-        public void ParseInvalidEscapeUriPathShouldThrow(string pattern, string segment)

```

Comment:
Are these now both valid? What is the behavior?

Reply:
Yes, both are valid for this step. These will throw an Unrecogonized Path Exception in the E2E scenario while binding. 

---
In reply to: [366144414](https://github.com/OData/odata.net/pull/1621#discussion_r366144414) [](ancestors = 366144414)